src/options.cpp

```diff
@@ -1458,9 +1458,15 @@ void options_manager::add_options_interface()
     "show", COPT_CURSES_HIDE );
 
     add( "EDGE_SCROLL", "interface", translate_marker( "Edge scrolling" ),
-         translate_marker( "If true, enables edge scrolling/panning with mouse when looking or peeking." ),
-         true, COPT_CURSES_HIDE
-       );
+    translate_marker( "Edge scrolling with the mouse." ), {
+        { "disabled", translate_marker( "Disabled" ) },
+        { "slow", translate_marker( "Slow" ) },
+        { "normal", translate_marker( "Normal" ) },
+        { "fast", translate_marker( "Fast" ) },
+        { "veryfast", translate_marker( "Very fast" ) }
```

Comment:
Please refrain from overusing strings. They get converted to `int`s in `mouse_edge_scrolling_speed` anyways, so why don't put those `int`s right here, instead of the strings? This would make the code more robust whilst obsoleting the `switch-case` there.

Reply:
Gah, that is exactly what I wanted (I initially went for `int_map`). Thanks!

Reply:
So what you suggest is something like this I guess (?):
```diff
-        { "disabled", translate_marker( "Disabled" ) },
-        { "slow", translate_marker( "Slow" ) },
-        { "normal", translate_marker( "Normal" ) },
-        { "fast", translate_marker( "Fast" ) },
-        { "veryfast", translate_marker( "Very fast" ) }
+        { 0, translate_marker( "Disabled" ) },
+        { 1, translate_marker( "Slow" ) },
+        { 2, translate_marker( "Normal" ) },
+        { 4, translate_marker( "Fast" ) },
+        { 8, translate_marker( "Very fast" ) }
```
(then we can use `get_option<int>( "EDGE_SCROLL" )` in place of `mouse_edge_scrolling_speed()`)

So then the option becomes an `int_map` type option (which, btw, is not used by any other option). For the user the values look like e.g. "0: Disabled" and "2: Normal" which exposes an internal value that they shouldn't meed to know about. And pressing Enter on the option opens an input prompt that probably doesn't behave as the user expect. And there are gaps in what can be inputted, e.g. 5 isn't valid and becomes the default (2).

The user should only see a string, so I think the current solution with a string option + `mouse_edge_scrolling_speed()` still makes most sense. Please let me know if I've missed something!

Reply:
> which, btw, is not used by any other option

Which means you're in position to adjust it to suit your needs. Furthermore, it would help eliminating string obsessions in other places (since they're essentially the same). The current string-based approach is not a merge-blocker, of course, but it's not the best imaginable solution either. Alternatively, you could use a simple `int` option for the speed of scrolling and let the user tune it as they want.

Reply:
Good point! I've adjusted `int_map` according to my needs.